src/orbital-helpers/semi_canonicalize.cc

```diff
@@ -402,32 +404,14 @@ void SemiCanonical::build_transformation_matrices(psi::SharedMatrix& Ua, psi::Sh
 }
 
 void SemiCanonical::transform_ints(psi::SharedMatrix& Ua, psi::SharedMatrix& Ub) {
-    psi::SharedMatrix Ca = wfn_->Ca();
-    psi::SharedMatrix Cb = wfn_->Cb();
-    psi::SharedMatrix Ca_new(Ca->clone());
-    psi::SharedMatrix Cb_new(Cb->clone());
-    Ca_new->gemm(false, false, 1.0, Ca, Ua, 0.0);
-    Cb_new->gemm(false, false, 1.0, Cb, Ub, 0.0);
-    Ca->copy(Ca_new);
-    Cb->copy(Cb_new);
-
     // Transform the integrals in the new basis
     print_h2("Integral Transformation to Semicanonical Basis");
-    ints_->retransform_integrals();
+    ints_->rotate_orbitals(Ua, Ub);
 }
 
 void SemiCanonical::back_transform_ints(psi::SharedMatrix& Ua, psi::SharedMatrix& Ub) {
-    psi::SharedMatrix Ca = wfn_->Ca();
-    psi::SharedMatrix Cb = wfn_->Cb();
-    psi::SharedMatrix Ca_new(Ca->clone());
-    psi::SharedMatrix Cb_new(Cb->clone());
-    Ca_new->gemm(false, true, 1.0, Ca, Ua, 0.0);
-    Cb_new->gemm(false, true, 1.0, Cb, Ub, 0.0);
-    Ca->copy(Ca_new);
-    Cb->copy(Cb_new);
-
     print_h2("Back Transformation of Semicanonical Integrals");
-    ints_->retransform_integrals();
+    ints_->rotate_orbitals(Ua->transpose(), Ub->transpose());
 }
```

Comment:
Both of these transformation functions seems a bit redundant now that we've implemented them in the integrals class.